src/main/scala/org/wikimedia/analytics/hdfstools/HDFSArchiver.scala

```diff
@@ -0,0 +1,237 @@
+     * Checks if a specific path exists in the file system, and checks that it is a directory.
+     */
+    def directoryExists(fs: FileSystem, path: Path): Boolean = {
+        if (!fs.exists(path)) {
+            log.error(s"Dir ${path.toString} does not exist.")
+            false
+        } else if (!fs.isDirectory(path)) {
+            log.error(s"Dir ${path.toString} is not a directory.")
+            false
+        } else {
+            true
+        }
+    }
+
+    /**
+     * Try to get the Path to the source file and checks that a there is single non empty file in the source directory
+     * (except the success file).
+     */
+    def getSourceFileFromDirectory(
+        fs: FileSystem,
+        sourceDirectory: Path,
+        expectedFilenameEnding: String,
+        checkDone: Boolean
+    ): Option[Path] = {
+        val files = listFilesInDir(fs, sourceDirectory)
+        if ((checkDone && files.length != 2) || (!checkDone && files.length != 1)) {
+            log.error(s"Wrong file count in ${sourceDirectory.toString}")
+            None
+        } else {
+            val sourceFile: Option[LocatedFileStatus] = files
+                .find {
+                    _.getPath.getName.endsWith(expectedFilenameEnding)
+                }
+            if (sourceFile.isEmpty) {
+                log.error(s"Missing source in ${sourceDirectory.toString} (ending in: $expectedFilenameEnding)")
+                None
+            } else if (sourceFile.get.getLen == 0) {
+                log.error(s"Empty source in ${sourceDirectory.toString} (ending in: $expectedFilenameEnding)")
+                None
+            } else {
+                Some(sourceFile.get.getPath)
+            }
+        }
+    }
+
+    /**
+     * Lists the files into a directory, and converts the iterator into a list for convenience.
+     */
+    def listFilesInDir(fs: FileSystem, dir: Path): List[LocatedFileStatus] = {
+        val result = ListBuffer[LocatedFileStatus]()
+        val iterator = fs.listFiles(dir, true)
+        while (iterator.hasNext) result += iterator.next
+        result.toList
+    }
+
+    def createParentFolder(fs: FileSystem, file: Path): Boolean = {
+        // This import has to happen after setting the umask mode
+        import org.apache.hadoop.fs.permission.FsPermission
+        fs.mkdirs(file.getParent, new FsPermission("777")) // Only restrict through umask.
+    }
+
+    def archiveSource(fs: FileSystem, sourceFile: Path, archiveFile: Path, archivePerms: String): Boolean = {
+        import org.apache.hadoop.fs.permission.FsPermission
+        fs.delete(archiveFile, false)
+        val result = fs.rename(sourceFile, archiveFile) &&
+            fs.delete(sourceFile.getParent, true) &&
+            fs.setPermission(archiveFile, new FsPermission(archivePerms)).equals() // "equals" is here to get a bool return.
+        if (result) {
+            log.info(s"Archive created: ${archiveFile.toString}")
+        }
+        result
+    }
+}
```

Comment:
There should be a newline at the end of the file

Reply:
Got it. Will add in the next commit